### PR TITLE
Feature/housekeeping

### PR DESCRIPTION
**Pull request by michsa**
- add actual 3.0 documentation to changelog
- fix moving checkbox bug
- Box improvements:
  - change modal/popup variant api from `variant` prop to `Box.Modal`/`Box.Popup` (for consistency with other components, eg. buttons)
  - kill `withPadding` HOC, move padding stuff directly into Button component
  - add `withAliasProps` HOC so we can have both `p` and `padding` props (and `m` + `margin` on Divider)

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR has lint errors. Please fix these and commit them.
      
<details>
  <summary>theme.js</summary>
  <p><a href="https://github.com/smartprocure/grey-vest/blob/feature/housekeeping/src/theme.js" target="_blank">src/theme.js</a></p>
  
  <strong>'jsx' is defined but never used.</strong>
  <code>Line 2: import { jsx } from '@emotion/core'</code>

  <strong>'coalesce' is defined but never used.</strong>
  <code>Line 5: import { coalesce } from './utils'</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Could not run `duti:fix` command successfully
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 582cf72b91f85895453fe00ed1aeb96824a2fbaa
</p>

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ab3637541363e3abba0ac5e03b025cc6bb2d961c
</p>

File: src/utils.js [hunk 1]
```diff
@@ -25,3 +25,11 @@ export let findKeys = _.curry((predicate, data) =>
 )
 
 export let coalesce = _.find(F.isNotNil)
+
+// ({ r: 'red', b: 'blue' g: 'green' }) -> Component<{ red, blue, green }> -> Component<{ red, blue, green, r, g, b }>
+export let withAliasProps = aliases => Component => props => (
```
Comment (daedalus28) [inline src/utils.js]: No need to reinvent the wheel:

https://github.com/acdlite/recompose/blob/master/docs/API.md#renameprops

File: src/Box.js [hunk 2]
```diff
@@ -1,17 +1,37 @@
...
+        _.join(' ')
+      )([coalesce([py, p]), coalesce([px, p])]),
     }}
     {...props}
   />
+))
+
+F.extendOn(
+  Box,
+  F.arrayToObject(
```
Comment (daedalus28) [inline src/Box.js]: I think this might be worse than the explicit approach

**Pull request merged**